ckan/lib/munge.py

```diff
@@ -119,6 +127,34 @@ def munge_filename(filename):
     return filename
 
 
+def munge_filename(filename):
+    ''' Tidies a filename
+
+    Keeps the filename extension.
+    Strips off any path on the front.
+    It can be useful to keep the extension (e.g. .csv) in a resource when the
+    format field set.
+    '''
+
+    # just get the filename ignore the path
+    path, filename = os.path.split(filename)
+    # clean up
+    filename = substitute_ascii_equivalents(filename)
+    filename = filename.lower().strip()
+    filename = re.sub(r'[^a-zA-Z0-9. -]', '', filename).replace(' ', '-')
+    # resize if needed but keep extension
+    name, ext = os.path.splitext(filename)
+    # limit overly long extensions
+    if len(ext) > 21:
+        ext = ext[:21]
+    # max/min size
+    ext_length = len(ext)
+    name = _munge_to_length(name, 3 - ext_length, 100 - ext_length)
```

Comment:
Good spot on the negatives. It happens to work ok, but just to be sure I've added a min value of 1.

Not sure I can do much about '.ext' without getting rid of splitext:
```
>>> os.path.splitext('.ext')
('.ext', '')
```
And I'm not convinced it is an edge case to necessarily worry about.